library/unwind/src/libunwind.rs

```diff
@@ -79,7 +79,7 @@ pub type _Unwind_Exception_Cleanup_Fn =
     extern "C" fn(unwind_code: _Unwind_Reason_Code, exception: *mut _Unwind_Exception);
 #[cfg_attr(
     all(feature = "llvm-libunwind", any(target_os = "fuchsia", target_os = "linux")),
-    link(name = "unwind", kind = "static")
+    link(name = "unwind", kind = "static", cfg(not(target_feature = "crt-static")))
```

Comment:
Could all the `#[link]` attributes be moved to a single place in `library\unwind\src\lib.rs`?
It's hard to understand what happens if they are scattered across multiple files.

Reply:
Also, without an explaining comment this logic looks pretty nonsensical.
"libunwind is linked statically, but only if libc is linked dynamically? what, shouldn't we do the opposite thing?"

Reply:
I revert this change. It does not work anyway until https://github.com/rust-lang/rust/issues/76020 is fixed.